src/web-map.js

```diff
@@ -447,6 +457,65 @@ export class WebMap extends HTMLMapElement {
     this.lon = this._map.getCenter().lng;
     this.zoom = this._map.getZoom();
   }
+
+  _addToHistory(){
+    if(this._traversalCall){
+      this._traversalCall = false;
+      return;
+    }
+    let mapLocation = this._map.getCenter();
+    let location ={
+      zoom:this._map.getZoom(),
+      lat:mapLocation.lat,
+      lng:mapLocation.lng,
+    };
+    this._historyIndex++;
+    this._history.push(location);
+  }
+
+  back(){
+    let mapEl = this,
+        history = mapEl._history;
+    if(mapEl._historyIndex > 0){
+      mapEl._historyIndex--;
+    }
+    let prev = history[mapEl._historyIndex];
+    mapEl._traversalCall = true;
+    mapEl.zoomTo(prev.lat,prev.lng,prev.zoom);
+  }
+
+  forward(){
+    let mapEl = this,
+        history = this._history;
+    if(mapEl._historyIndex < history.length -1){
+      mapEl._historyIndex++;
+    }
+    let next = history[this._historyIndex];
+    mapEl._traversalCall = true;
+    mapEl.zoomTo(next.lat,next.lng,next.zoom);
+  }
+
+  reload(){
+    let mapEl = this,
+        initialLocation = mapEl._history.shift();
+    mapEl._history = [initialLocation];
+    mapEl._historyIndex = -1;
+    mapEl._traversalCall = true;
+    mapEl.zoomTo(initialLocation.lat,initialLocation.lng,initialLocation.zoom);
+  }
+
+  viewSource(){
+    let mapHTML = this.outerHTML,
```

Comment:
I think that since we insert the `<style></style>` as a child of the <map> element, we should not show it, or any other DOM nodes that *we* create... it's supposed to be the source *before* script has run, is what I'm saying.

Reply:
That's a good point, I'll add that change

Reply:
Yeah, it's not going to be perfect I think because I believe the attributes are reflecting property changes, but still.